tests/beaker_tests/acl/test_acl.rb

```diff
@@ -0,0 +1,271 @@
+#   be merged with title values and override any duplicates. If omitted,
+#   :title_pattern will be set to 'id'.
+# tests[id][:af] - (Optional) defines the address-family values.
+#   Must use :title_pattern if :af is not specified. Useful for testing mixed
+#   title/af manifests
+#
+tests['default_properties_ipv4'] = {
+  title_pattern:  'ipv4 beaker1',
+  manifest_props: "
+    stats_per_entry                  => 'false',
+    fragments                       => 'default',
+  ",
+  resource_props: {
+    'stats_per_entry' => 'false',
+    'fragments'      => '',
+  },
+}
+
+tests['default_properties_ipv6'] = {
+  title_pattern:  'ipv6 beaker_2',
+  manifest_props: "
+    stats_per_entry                  => 'false',
+    fragments                       => 'default',
+  ",
+  resource_props: {
+    'stats_per_entry' => 'false',
+    'fragments'      => '',
+  },
+}
+
+tests['non_default_properties_ipv4'] = {
+  title_pattern:  'ipv4 beaker_3',
+  manifest_props: "
+    stats_per_entry                  => 'true',
+    fragments                       => 'permit-all',
+  ",
+  resource_props: {
+    'stats_per_entry' => 'true',
+    'fragments'      => 'permit-all',
+  },
+}
+
+tests['non_default_properties_ipv6'] = {
+  title_pattern:  'ipv6 beaker_4',
+  manifest_props: "
+    stats_per_entry                  => 'true',
+    fragments                       => 'deny-all',
+  ",
+  resource_props: {
+    'stats_per_entry' => 'true',
+    'fragments'      => 'deny-all',
+  },
+}
+
+
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+
+# Full command string for puppet resource command
+def puppet_resource_cmd
+  cmd = UtilityLib::PUPPET_BINPATH + 'resource cisco_acl'
+  UtilityLib.get_namespace_cmd(agent, cmd, options)
+end
+
+def build_manifest_acl(tests, id)
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    tests[id][:resource] = {}
+  else
+    state = 'ensure => present,'
+    manifest = tests[id][:manifest_props]
+    tests[id][:resource] = tests[id][:resource_props]
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  logger.debug("build_manifest_acl :: title_pattern:\n" +
+               tests[id][:title_pattern])
+  tests[id][:manifest] = "cat <<EOF >#{UtilityLib::PUPPETMASTER_MANIFESTPATH}
+  node 'default' {
+    cisco_acl { '#{tests[id][:title_pattern]}':
+      #{state}
+      #{manifest}
+    }
+  }
+EOF"
+end
+
+def test_harness_acl(tests, id)
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+  tests[id][:resource_cmd] = puppet_resource_cmd
+  tests[id][:desc] += " [ensure => #{tests[id][:ensure]}]"
+
+  # Build the manifest for this test
+  build_manifest_acl(tests, id)
+
+  # FUTURE
+  # test_harness_common(tests, id)
+
+  test_manifest(tests, id)
+  test_resource(tests, id)
+  test_idempotence(tests, id)
```

Comment:
I think you should be able to use `test_harness_common()` here instead of these 3 calls.